Dalamud/Interface/ManagedFontAtlas/Internals/ImGuiDockNodeUpdateForceFont.cs

```diff
@@ -0,0 +1,78 @@
+using System.Diagnostics;
+using System.Linq;
+
+using Dalamud.Hooking;
+using Dalamud.Interface.Internal;
+using Dalamud.Interface.Utility;
+
+namespace Dalamud.Interface.ManagedFontAtlas.Internals;
+
+/// <summary>
+/// Forces ImGui::RenderWindowDecorations to use the default font.
+/// Fixes dock node draw using shared data across different draw lists.
+/// TODO: figure out how to synchronize ImDrawList::_Data and ImDrawList::Push/PopTextureID across different instances.
+/// It might be better to just special-case that particular function,
+/// as no other code touches ImDrawList that is irrelevant to the global shared state,
+/// with the exception of Dock... functions which are called from ImGui::NewFrame,
+/// which are guaranteed to use the global default font.
+/// </summary>
+[ServiceManager.EarlyLoadedService]
+internal class ImGuiRenderWindowDecorationsForceFont : IServiceType, IDisposable
+{
+    private const int CImGuiRenderWindowDecorationsOffset = 0x461B0;
+    private const int CImGuiWindowDockIsActiveOffset = 0x401;
```

Comment:
Will these offsets need changing if/when we update our cimgui instance?

Reply:
Yeah, along with the constants at `DalamudIme.cs`